CI: data race in endpoint.TestIncrementalUpdatesDuringPolicyGeneration

## CI failure

```
==================
WARNING: DATA RACE
Read at 0x00c00038e6ff by goroutine 22:
  github.com/cilium/cilium/pkg/endpoint.TestIncrementalUpdatesDuringPolicyGeneration()
      /home/tklauser/go/src/github.com/cilium/cilium/pkg/endpoint/policy_test.go:170 +0x183e
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1851 +0x44

Previous write at 0x00c00038e6ff by goroutine 24:
  github.com/cilium/cilium/pkg/endpoint.TestIncrementalUpdatesDuringPolicyGeneration.func2()
      /home/tklauser/go/src/github.com/cilium/cilium/pkg/endpoint/policy_test.go:128 +0x232

Goroutine 22 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1851 +0x8f2
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:2279 +0x85
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1792 +0x225
  testing.runTests()
      /usr/local/go/src/testing/testing.go:2277 +0x96c
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:2142 +0xeea
  main.main()
      _testmain.go:101 +0x164

Goroutine 24 (finished) created at:
  github.com/cilium/cilium/pkg/endpoint.TestIncrementalUpdatesDuringPolicyGeneration()
      /home/tklauser/go/src/github.com/cilium/cilium/pkg/endpoint/policy_test.go:114 +0x1512
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1792 +0x225
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1851 +0x44
==================
--- FAIL: TestIncrementalUpdatesDuringPolicyGeneration (2.62s)
```

Seen in local testing, can easily be reproduced using:

```
go test -race -c ./pkg/endpoint && stress ./endpoint.test -test.run=IncrementalUpdatesDuringPolicyGeneration
```

cc @squeed

--- a/pkg/endpoint/policy_test.go
+++ b/pkg/endpoint/policy_test.go
@@ -8,6 +8,7 @@ import (
 	"fmt"
 	"math/rand/v2"
 	"sync"
+	"sync/atomic"
 	"testing"
 	"time"
 
@@ -108,7 +109,7 @@ func TestIncrementalUpdatesDuringPolicyGeneration(t *testing.T) {
 	// Track all IDs we allocate so we can validate later that we never miss any
 	checkMutex := lock.Mutex{}
 	allocatedIDs := make(sets.Set[identity.NumericIdentity], testfactor)
-	done := false
+	done := atomic.Bool{}
 
 	// simulate ipcache churn: continuously allocate IDs and push them to the policy engine.
 	go func() {
@@ -125,7 +126,7 @@ func TestIncrementalUpdatesDuringPolicyGeneration(t *testing.T) {
 			checkMutex.Unlock()
 
 		}
-		done = true
+		done.Store(true)
 	}()
 
 	stats := new(regenerationStatistics)
@@ -167,7 +168,7 @@ func TestIncrementalUpdatesDuringPolicyGeneration(t *testing.T) {
 
 		checkMutex.Unlock()
 
-		if done {
+		if done.Load() {
 			break
 		}
 	}

